src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -545,33 +578,65 @@ final case class AgreementApiServiceImpl(
-        agreement = agreement,
-        eService = eService,
-        consumer = consumer,
-        producer = producer,
-        seed = seed
-      )
-      _        <- authorizationManagementService
-        .updateStateOnClients(
-          eServiceId = agreement.eserviceId,
-          consumerId = agreement.consumerId,
-          agreementId = agreement.id,
-          state = toClientState(newState)
+    def createContractAndSendMail(agreement: AgreementManagement.Agreement, seed: UpdateAgreementSeed): Future[Unit] =
+      for {
+        producer <- tenantManagementService.getTenant(agreement.producerId)
+        _        <- agreementContractCreator.create(
+          agreement = agreement,
+          eService = eService,
+          consumer = consumer,
+          producer = producer,
+          seed = seed
         )
-      _        <- sendActivationEnvelope(agreement, producer, consumer, eService)
-    } yield ()
+        _        <- sendActivationEnvelope(agreement, producer, consumer, eService)
+      } yield ()
 
     for {
       uid    <- getUidFutureUUID(contexts)
       stamps <- calculateStamps(newState, Stamp(uid, offsetDateTimeSupplier.get()))
-      updateSeed = getUpdateSeed(newState, agreement, stamps)
+      updateSeed = getUpdateSeed(stamps)
       updated <- agreementManagementService.updateAgreement(agreement.id, updateSeed)
-      _       <- validateResultState(newState)
-      _       <-
-        if (updated.state == AgreementManagement.AgreementState.ACTIVE) performActivation(updated, updateSeed)
+
+      agreements <- agreementManagementService
+        .getAgreements(
+          consumerId = agreement.consumerId.toString.some,
+          eserviceId = agreement.eserviceId.toString.some,
+          states = List(AgreementManagement.AgreementState.ACTIVE, AgreementManagement.AgreementState.SUSPENDED)
+        )
+        .map(_.filterNot(_.id == agreement.id))
+
+      _ <-
+        if (
+          newState == AgreementManagement.AgreementState.ACTIVE || newState == AgreementManagement.AgreementState.SUSPENDED
+        )
+          Future
+            .traverse(agreements)(a =>
+              agreementManagementService.updateAgreement(
+                a.id,
+                UpdateAgreementSeed(
+                  state = AgreementManagement.AgreementState.ARCHIVED,
+                  certifiedAttributes = a.certifiedAttributes,
+                  declaredAttributes = a.declaredAttributes,
+                  verifiedAttributes = a.verifiedAttributes,
+                  stamps = a.stamps.copy(archiving = Stamp(uid, offsetDateTimeSupplier.get()).some)
+                )
+              )
+            )
+            .map(_ => ())
+        else Future.unit
+
+      // * Se lo stato non è attivo a causa dei flags ritorna un errore applicativo
+      // * Per questo viene fatto il controllo a valle
+      _ <- validateResultState(newState)
+      _ <- authorizationManagementService.updateStateOnClients(
```

Comment:
after sleeping on it, should this be performed only if the new state is `ACTIVE` or `SUSPENDED`?
If the new agreement is not in a "operational" state, clients should keep the current configuration (for example if the resulting state is `PENDING`).
Does this make sense?

Reply:
Hmm, indeed it does, that's probably was was there in the first place

Reply:
Updated